### PR TITLE
Refactor participation lambdas

### LINKED ISSUE #74: [HexArch] Refactor participation lambdas

### PR DESCRIPTION
**Pull request by rubenhorn**
Closes #74

### CODE REVIEW

File: pkg/repository/aws/crowdactionParticipationManager.go
Comment (edreinoso) [inline pkg/repository/aws/crowdactionParticipationManager.go:1]: Instead of creating a new file for the crowdactionParticipationManager, I would try to put this method in the dynamo.go file. Maybe I misunderstood the concept of the repositories, but I thought they were supposed to be revolving around the AWS services as opposed to the business interactions. Ultimately it'd an interesting discussion to have.

File: pkg/repository/aws/participationManager.go
Comment (edreinoso) [inline pkg/repository/aws/participationManager.go:1]: Same comment from the crowdactionParticipationManager applies to the this file.

File: pkg/handler/aws/participationAggregation/aggregation.go
Comment (edreinoso) [inline pkg/handler/aws/participationAggregation/aggregation.go:4]: I don't understand how this file may be used in this pull request. I thought the controller was mostly part of the old architecture, but I am not sure.

**Review by edreinoso — COMMENTED** [review]
I understood most of the changes, and for me they were to be approved if we had a discussion about the usage of the repository/aws/ directory purpose.

File: pkg/repository/aws/crowdactionParticipationManager.go
Comment (rubenhorn) [reply pkg/repository/aws/crowdactionParticipationManager.go:1]: I was thinking about it more in terms of DDD (https://medium.com/@pererikbergman/repository-design-pattern-e28c0f3e4a30). @mrsoftware what was your initial idea?  
Maybe we could use generic repositories for the database, but we should decide it once and then apply it consistently.

File: pkg/handler/aws/participationAggregation/aggregation.go
Comment (rubenhorn) [reply pkg/handler/aws/participationAggregation/aggregation.go:4]: This is used by the handler in `./main.go`.  
I can merge those two files if you want, but I thought it made the main file clearer to identify as the lambda handler.  
As you can see `func NewParticipationAggregationHandler` now uses the repository and service.

File: pkg/repository/aws/crowdactionParticipationManager.go
Comment (edreinoso) [reply pkg/repository/aws/crowdactionParticipationManager.go:1]: Oh now I see what you were trying to think of. I particularly liked this passage of the article

"I started with one giant DAO handling all my different models. Even if this humungous big class was convenient it was highly clumsy and hard to modify, maintain and almost impossible to reuse."

I'm not very picky with either approach, I just want to have clarification so that we can have a cohesive code base.

If we decide this ought to be the way, then this pull request would certainly be approved on my side.

File: pkg/handler/aws/participationAggregation/aggregation.go
Comment (edreinoso) [reply pkg/handler/aws/participationAggregation/aggregation.go:4]: Nah it's completely fine. I just didn't understand in the beginning, but now that you are pointing it out it makes sense

File: pkg/repository/aws/crowdactionParticipationManager.go
Comment (rubenhorn) [reply pkg/repository/aws/crowdactionParticipationManager.go:1]: I think this relates to the recent comments by @mrsoftware on #119.

Comment (edreinoso) [reply pkg/repository/aws/crowdactionParticipationManager.go:1]: All righty then!

**Review by edreinoso — APPROVED** [review]
All comments have been addressed properly

**Pull request merged**
